app/src/org/odk/collect/android/views/ResizingImageView.java

```diff
@@ -157,41 +158,22 @@ public void setFullScreen(){
         }
     }
 
-    public Pair<Integer,Integer> getWidthHeight(int widthMeasureSpec, int heightMeasureSpec, Drawable drawable, double scaleFactor){
+    public Pair<Integer,Integer> getWidthHeight(int widthMeasureSpec, int heightMeasureSpec, double scaleFactor){
 
         int wMode = MeasureSpec.getMode(widthMeasureSpec);
         int hMode = MeasureSpec.getMode(heightMeasureSpec);
 
-        // Calculate the most appropriate size for the view. Take into
-        // account minWidth, minHeight, maxWith, maxHeigh and allowed size
-        // for the view, then scale by the scaleFactor
-
         int maxWidth = wMode == MeasureSpec.AT_MOST
                 ? Math.min(MeasureSpec.getSize(widthMeasureSpec), mMaxWidth)
-                        : mMaxWidth;
-                int maxHeight = hMode == MeasureSpec.AT_MOST
-                        ? Math.min(MeasureSpec.getSize(heightMeasureSpec), mMaxHeight)
-                                : mMaxHeight;
-
-                        float dWidth = dipToPixels(getContext(), drawable.getIntrinsicWidth());
-                        float dHeight = dipToPixels(getContext(), drawable.getIntrinsicHeight());
-                        float ratio = (dWidth) / dHeight;
-
-                        int width = (int) Math.min(Math.max(dWidth, getSuggestedMinimumWidth()), maxWidth);
-                        int height = (int) (width / ratio);
-
-                        height = Math.min(Math.max(height, getSuggestedMinimumHeight()), maxHeight);
-                        width = (int) (height * ratio);
-
-                        if (width > maxWidth) {
-                            width = maxWidth;
-                            height = (int) (width / ratio);
-                        }
+                    : mMaxWidth;
 
-                        Pair<Integer,Integer> mPair = new Pair<Integer,Integer>(new Double(width * scaleFactor).intValue(), new Double(height * scaleFactor).intValue());
+        int maxHeight = hMode == MeasureSpec.AT_MOST
+                ? Math.min(MeasureSpec.getSize(heightMeasureSpec), mMaxHeight)
+                    : mMaxHeight;
 
+        Pair<Integer,Integer> mPair = new Pair<Integer,Integer>(new Double(maxWidth * scaleFactor).intValue(), new Double(maxHeight * scaleFactor).intValue());
 
-                        return mPair;
+        return mPair;
```

Comment:
It is cleaner to return an expression instead of assigning it a variable and then immediately returning that variable.